pvlib/iotools/pvgis.py

```diff
@@ -0,0 +1,156 @@
+"""
+Get, read, and parse data from `PVGIS <https://ec.europa.eu/jrc/en/pvgis>`_.
+
+For more information, see the following links:
+* `Interactive Tools <https://re.jrc.ec.europa.eu/pvg_tools/en/tools.html>`_
+* `Data downloads <https://ec.europa.eu/jrc/en/PVGIS/downloads/data>`_
+* `User manual docs <https://ec.europa.eu/jrc/en/PVGIS/docs/usermanual>`_
+
+More detailed information about the API for TMY and hourly radiation are here:
+* `TMY <https://ec.europa.eu/jrc/en/PVGIS/tools/tmy>`_
+* `hourly radiation
+  <https://ec.europa.eu/jrc/en/PVGIS/tools/hourly-radiation>`_
+* `daily radiation <https://ec.europa.eu/jrc/en/PVGIS/tools/daily-radiation>`_
+* `monthly radiation
+  <https://ec.europa.eu/jrc/en/PVGIS/tools/monthly-radiation>`_
+"""
+import io
+import tempfile
+import requests
+import pandas as pd
+from pvlib.iotools import read_epw
+
+URL = 'https://re.jrc.ec.europa.eu/api/'
+
+
+def get_pvgis_tmy(lat, lon, outputformat='json', usehorizon=True,
+                  userhorizon=None, startyear=None, endyear=None):
+    """
+    Get TMY data from PVGIS. For more information see documentation for PVGIS
+    `TMY tools <https://ec.europa.eu/jrc/en/PVGIS/tools/tmy>`_
+
+    Parmaeters
+    ----------
+    lat : float
+        Latitude in degrees north
+    lon : float
+        Longitude in dgrees east
+    outputformat : string [default 'json']
+        Must be in ``['csv', 'basic', 'epw', 'json']``. See link for more info.
+    usehorizon : bool [default True]
+        include effects of horizon
+    userhorizon : list of float [default None]
+        elevation of horizon in degrees at eight cardinal directions clockwise
```

Comment:
PVGIS documents say that a list of any length is accepted, and interpreted as horizon elevation at equally spaced azimuth clockwise from north.